subsys/net/ip/net_private.h

```diff
@@ -214,14 +223,37 @@ static inline void net_hexdump(const char *str, const u8_t *packet,
 	}
 }
 
-/* Hexdump from all fragments */
-static inline void net_hexdump_frags(const char *str, struct net_pkt *pkt)
+static inline void net_hexdump(const char *str,
+			       const u8_t *packet, size_t length)
 {
+	if (!length) {
+		SYS_LOG_DBG("%s zero-length packet", str);
+		return;
+	}
+
```

Comment:
Print header (the str variable) here too.

Reply:
indeed